hal/src/rtl872x/usart_hal.cpp

```diff
@@ -717,6 +826,10 @@ class Usart {
         uart->dataInFlight(true /* commit */, true /* cancel */);
         uart->receiving_ = false;
         uart->startReceiver();
+        BaseType_t yield = pdFALSE;
+        if (xEventGroupSetBitsFromISR(uart->evGroup_, HAL_USART_PVT_EVENT_READABLE, &yield) != pdFAIL) {
```

Comment:
This doesn't look right. This interrupt handler only generates at the end of the full DMA transfer. This event should be generated once we receive even just 1 byte.

Reply:
In that case, we need to enable rx interrupt when using  DMA mode and only set this event bit in the ISR. Sounds good?

Reply:
Yes, that's the idea (it's the same on Gen 3). And disable immediately once it happens. Also this interrupt should ideally be triggered on transfer out of UART FIFO not on when the byte appears in UART FIFO. + We need to syncup the data in flight on getting unblocked by this event.